app/assets/javascripts/whole_pot_calculator.js

```diff
@@ -0,0 +1,212 @@
+(function() {
+  'use strict';
+
+  function WholePotCalculator(income, pot) {
+    this.income = income;
+    this.pot = pot;
+
+    this.standardPersonalAllowance = 10600;
+
+    this.basicRateUpperLimit = 31785;
+    this.basicRateTax = 0.2;
+
+    this.higherRateUpperLimit = 150000;
+    this.higherRateTax = 0.4;
+
+    this.additionalRateTax = 0.45;
+
+    this.taxablePotPortion = 0.75;
+  }
+
+  WholePotCalculator.prototype.round = function(num) {
+    return Math.round(num * 100) / 100;
+  };
+
+  WholePotCalculator.prototype.personalAllowanceFor = function(totalIncome) {
+    if(totalIncome <= 100000) {
+      return this.standardPersonalAllowance;
+    } else {
+      var reduction = Math.min(Math.floor((totalIncome - 100000) / 2), this.standardPersonalAllowance);
+
+      return this.standardPersonalAllowance - reduction;
+    }
+  };
+
+  WholePotCalculator.prototype.marginalTaxForIncomeWithAllowance = function(income, allowance) {
```

Comment:
Strictly it probably shouldn't be part of the public API but it was the simplest thing I could focus on at the start of development